src/appleseed.studio/resources/shaders/scene.frag

```diff
@@ -28,37 +28,26 @@
 #version 330
 #extension GL_ARB_separate_shader_objects : enable
 
-layout(location = 0) in vec3 v_world_pos;
-layout(location = 1) in vec3 v_norm;
-
-uniform vec3 u_camera_pos;
-
-const vec3 LIGHT_POS = vec3(0.0, 0.0, 0.0);
-const vec3 LIGHT_COLOR = vec3(1.0);
-const vec3 AMBIENT_COLOR = vec3(0.05);
+layout(location = 0) in vec3 frag_pos;
+layout(location = 1) in vec3 frag_norm;
 
 const vec3 MATERIAL_BASE_COLOR = vec3(0.8, 0.8, 0.8);
 const float MATERIAL_DIFFUSE_FACTOR = 0.4;
 const float MATERIAL_SPECULAR_FACTOR = 0.15;
-const float MATERIAL_METALNESS_FACTOR = 0.0;
-const float MATERIAL_SHININESS = 80.0;
+const float MATERIAL_SHININESS = 32.0;
 
 out vec4 Target0;
 
 void main()
 {
-    vec3 acc = vec3(0.0);
-
-    vec3 N = normalize(v_norm);
-    vec3 L = normalize(LIGHT_POS - v_world_pos);
-    vec3 V = normalize(u_camera_pos - v_world_pos);
-    vec3 H = normalize(L + V);
+    vec3 view_dir = normalize(-frag_pos);
+    vec3 reflect_dir = reflect(-view_dir, frag_norm);  
```

Comment:
Remove trailing whitespaces.

Reply:
I'm kind of a noob when it comes to GPU shader writing, but since `frag_norm` is interpolated, doesn't it require a normalization here?

Reply:
Indeed, it does. I didn't normalize it because I was looking for a flat look. But it's not correct. The flat look comes from the duplication of the normals.